core/storage/auth/gae_models_test.py

```diff
@@ -72,6 +78,80 @@ def test_get_deletion_policy_is_delete_at_end(self) -> None:
             auth_models.UserAuthDetailsModel.get_deletion_policy(),
             base_models.DELETION_POLICY.DELETE_AT_END)
 
+    def test_get_field_names_for_takeout(self) -> None:
+        expected_dict = {
+            'parent_user_id': 'parent_username'
+        }
+        self.assertEqual(
+            auth_models.UserAuthDetailsModel
+                .get_field_names_for_takeout(),
+            expected_dict)
+
+    def test_get_model_association_to_user(self) -> None:
+        self.assertEqual(
+            auth_models.UserAuthDetailsModel.
+                get_model_association_to_user(),
+            base_models.MODEL_ASSOCIATION_TO_USER.ONE_INSTANCE_PER_USER)
+
+    def test_get_export_policy(self) -> None:
+        expected_export_policy_dict = {
+            'created_on': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'deleted': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'last_updated': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'gae_id': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'firebase_auth_id': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'parent_user_id': base_models.EXPORT_POLICY.EXPORTED,
+        }
+        self.assertEqual(
+            auth_models.UserAuthDetailsModel
+                .get_export_policy(),
+            expected_export_policy_dict)
+
+    def test_export_data_trivial(self) -> None:
+        """Trivial test of export_data functionality."""
+
+        exported_model = (
+            auth_models.UserAuthDetailsModel.export_data(
+                self.NONEXISTENT_USER_ID))
+
+        self.assertEqual(exported_model, {})
+
+    def test_export_data_nontrivial(self) -> None:
+        user_auth_model = (
+            auth_models.UserAuthDetailsModel
+                .get_by_id(self.PROFILE_2_ID))
+        self.assertIsNotNone(user_auth_model)
+
+        # The parent_user_id should exist to fetch the
+        # parent_model further on in the test.
```

Comment:
If this is the case then don't you need to do a stronger check here, like self.assertEqual(parent_user_id, self.USER_ID)? Otherwise, there is no connection to the subsequent model.

Reply:
Done!